### PR TITLE
[browserstack-service] Add more CIs

### PR DESCRIPTION
**Pull request by samarsault**
## Proposed changes
- Support more CI providers in browserstack-service

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/util.ts [hunk 1]
```diff
@@ -566,7 +566,7 @@ export function getCiInfo () {
         }
     }
     // CircleCI
-    if (env.CI === 'true' && env.CIRCLECI === 'true') {
+    if (isTrue(env.CI) && isTrue(env.CIRCLECI)) {
```
Comment (erwinheitzman) [inline packages/wdio-browserstack-service/src/util.ts:569]: let's not do this as for some comparisons something else than 'true' is used and the change is not required for the PR for it to work.

File: packages/wdio-browserstack-service/src/util.ts [hunk 2]
```diff
@@ -638,14 +638,140 @@ export function getCiInfo () {
...
+    if (isTrue(env.APPVEYOR)) {
+        return {
+            name: 'Appveyor',
+            build_url: `${env.APPVEYOR_URL}/project/${env.APPVEYOR_ACCOUNT_NAME}/${env.APPVEYOR_PROJECT_SLUG}/builds/${env.APPVEYOR_BUILD_ID}`,
+            job_name: env.APPVEYOR_JOB_NAME,
+            build_number: env.APPVEYOR_BUILD_NUMBER
+        }
+    }
+    // Azure CI
+    if (env.AZURE_HTTP_USER_AGENT && env.TF_BUILD) {
```
Comment (erwinheitzman) [inline packages/wdio-browserstack-service/src/util.ts:659]: Already defined at line 641

File: packages/wdio-browserstack-service/src/util.ts [hunk 3]
```diff
@@ -638,14 +638,140 @@ export function getCiInfo () {
...
+            build_url: env.WERCKER_BUILD_URL,
+            job_name: env.WERCKER_MAIN_PIPELINE_STARTED ? 'Main Pipeline' : null,
+            build_number: env.WERCKER_GIT_COMMIT
+        }
+    }
+    // Google Cloud
+    if (env.GCP_PROJECT || env.GCLOUD_PROJECT || env.GOOGLE_CLOUD_PROJECT) {
+        return {
+            name: 'Google Cloud',
+            build_url: null,
```
Comment (erwinheitzman) [inline packages/wdio-browserstack-service/src/util.ts:698]: Why are the url's set to null?

**Review by erwinheitzman — CHANGES_REQUESTED** [review]
Thanks for adding these, could you sign the CLA?

File: packages/wdio-browserstack-service/src/util.ts (hunk 1 above)
Comment (samarsault) [reply packages/wdio-browserstack-service/src/util.ts:569]: I'm not sure I understand. Can you give an example where the earlier change works but this doesn't?

File: packages/wdio-browserstack-service/src/util.ts (hunk 2 above)
Comment (samarsault) [reply packages/wdio-browserstack-service/src/util.ts:659]: That one is for Team Foundation, this is for Azure CI. [Ref](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/util/ci_provider.js#L23-L34) from cypress

File: packages/wdio-browserstack-service/src/util.ts (hunk 3 above)
Comment (samarsault) [reply packages/wdio-browserstack-service/src/util.ts:698]: Left it null in cases where that info is not available

**Comment by christian-bromann** [comment]
It seems like some of the unit tests are failing that are related to this change 🤔

File: packages/wdio-browserstack-service/src/util.ts (hunk 2 above)
Comment (erwinheitzman) [reply packages/wdio-browserstack-service/src/util.ts:659]: I see, in that case we will have to add TF_BUILD_BUILDNUMBER or there will by no distinction between the two ;)

File: packages/wdio-browserstack-service/src/util.ts (hunk 1 above)
Comment (erwinheitzman) [reply packages/wdio-browserstack-service/src/util.ts:569]: The code is working fine, that wasn't my point. Looking back at it, I don't really mind the change so I'll resolve this :)

File: packages/wdio-browserstack-service/src/util.ts (hunk 2 above)
Comment (samarsault) [reply packages/wdio-browserstack-service/src/util.ts:659]: Added

**Comment by samarsault** [comment]
@christian-bromann Unit tests are fixed

**Review by christian-bromann — APPROVED** [review]
LGTM 👍 

@erwinheitzman good to go?

**Review by erwinheitzman — APPROVED** [review]
LGTM, thank you for your contribution 👍

**Pull request merged**